litert/vendors/qualcomm/core/wrappers/op_wrapper.cc

```diff
@@ -34,6 +35,30 @@ OpWrapper& OpWrapper::operator=(const OpWrapper& other) {
   return *this;
 }
 
+bool OpWrapper::operator==(const OpWrapper& other) const {
+  if (op_code_ != other.op_code_) return false;
+  if (!IsStrEq(type_name_, other.type_name_)) return false;
+
+  if (input_tensors_.size() != other.input_tensors_.size()) return false;
+  if (!std::equal(
+          input_tensors_.begin(), input_tensors_.end(),
+          other.input_tensors_.begin(),
+          [](const auto& a, const auto& b) { return a.get() == b.get(); })) {
+    return false;
+  }
+
+  if (output_tensors_.size() != other.output_tensors_.size()) return false;
+  if (!std::equal(
+          output_tensors_.begin(), output_tensors_.end(),
+          other.output_tensors_.begin(),
+          [](const auto& a, const auto& b) { return a.get() == b.get(); })) {
+    return false;
+  }
```

Comment:
`std::equal` verifies whether std::distance(first1, last1) and std::distance(first2, last2) are equal. Could we consider updating the implementation as follows?
```suggestion
  if (!std::equal(
          input_tensors_.begin(), input_tensors_.end(),
          other.input_tensors_.begin(), other.input_tensors_.end()
          [](const auto& a, const auto& b) { return a.get() == b.get(); })) {
    return false;
  }

  if (!std::equal(
          output_tensors_.begin(), output_tensors_.end(),
          other.output_tensors_.begin(), other.output_tensors_.end(),
          [](const auto& a, const auto& b) { return a.get() == b.get(); })) {
    return false;
  }
```

Reply:
Thanks for the comment! This is a good optimization we should apply, updated in the latest commit.